backend/app/controllers/eventsController/db/onOrderChange.js

```diff
@@ -1,38 +1,43 @@
+const { Orders } = require('../../../db/models');
 
 module.exports = async (mongoEvent, io) => {
-    if (mongoEvent.operationType === "insert") {
-        const ownerRoomName = `${mongoEvent.fullDocument.restaurantId}:owner`;
-        const tabRoomName = `${mongoEvent.fullDocument.restaurantId}:${mongoEvent.fullDocument.tabId}`;
-        //todo remove duplicate code from here. make a function that returns 
-        //the order and use it here and in controllers
-        const order = await Orders.findById(mongoEvent.fullDocument._id)
-        .populate('items')
-        .populate({
-          path: 'tabId',
-          populate: {
-            path: 'tableId'
-          }
-        }).exec();
+	if (mongoEvent.operationType === 'insert') {
+		const ownerRoomName = `${mongoEvent.fullDocument.restaurantId}:owner`;
+		const tabRoomName = `${mongoEvent.fullDocument.restaurantId}:${mongoEvent.fullDocument.tabId}`;
+		//todo remove duplicate code from here. make a function that returns 
+		//the order and use it here and in controllers
+		const order = await Orders.findById(mongoEvent.fullDocument._id)
+			.populate({
+				path: 'items',
+				populate: {
+					path: 'recipe'
+				}
+			})
+			.populate({
+				path: 'tabId',
+				populate: {
+					path: 'tableId'
+				}
+			}).exec();
 
-        io.to(ownerRoomName).emit("order:new", { order });
-        console.log(`New order sent in room: "${ownerRoomName}"`);
+		io.to(ownerRoomName).emit('order:new', { order });
+		console.log(`New order sent in room: "${ownerRoomName}"`);
 
-        io.to(tabRoomName).emit("order:new", { order });
-        console.log(`New order sent in room: "${tabRoomName}"`);
-    }
+		io.to(tabRoomName).emit('order:new', { order });
+		console.log(`New order sent in room: "${tabRoomName}"`);
+	}
 
-    if (mongoEvent.operationType === "update") {
-        const order = await Orders.findById(mongoEvent.documentKey._id);
-        const ownerRoomName = `${order.restaurantId}:owner`;
-        const tabRoomName = `${order.restaurantId}:${order.tabId}`;
+	if (mongoEvent.operationType === 'update') {
+		const order = await Orders.findById(mongoEvent.documentKey._id);
+		const ownerRoomName = `${order.restaurantId}:owner`;
+		const tabRoomName = `${order.restaurantId}:${order.tabId}`;
         
-        io.to(ownerRoomName).emit("order:update", { id: order._id, status: order.status });
-        console.log(`New order update sent in room: "${ownerRoomName}"`);
+		io.to(ownerRoomName).emit('order:update', { id: order._id, status: order.status });
+		console.log(`New order update sent in room: "${ownerRoomName}"`);
 
-        io.to(tabRoomName).emit("order:update", { id: order._id, status: order.status });
-        console.log(`New order update sent in room: "${tabRoomName}"`);
+		io.to(tabRoomName).emit('order:update', { id: order._id, status: order.status });
+		console.log(`New order update sent in room: "${tabRoomName}"`);
         
 
-    }
+	}
 };
\ No newline at end of file
```

Comment:
The code patch is an event handler that listens to MongoDB change stream events and emits order updates to owner and table rooms via socket.io. Here are some improvement suggestions and bug risks to consider:

1. Improve function organization: The duplicate code comment suggests a potential refactoring opportunity to extract a reusable function for retrieving orders. It would be good to separate this functionality into a standalone controller or service.

2. Use consistent quotes: The code mixes single- and double-quotes for string literals. It's better to pick one style and use it consistently throughout the codebase.

3. Add error handling: The code does not include any error handling logic. It's important to handle errors that may occur during database operations to prevent unexpected crashes and provide better feedback to end-users.

4. Avoid magic strings: The code relies on hard-coded strings such as "insert" and "update" to determine the operation type. It would be better to define constants or enums to minimize the risk of typos and make the code more readable.

5. Potential bug risk: In the update case, the code retrieves the updated order using `mongoEvent.documentKey._id`, which assumes that the `_id` field is present in the change stream event. If this assumption is incorrect, the code will fail.

6. Consider using async/await: The format of the code is currently using callbacks with `exec()`. Async/await could simplify the code and make it easier to read.

7. Logging improvements: Consider adding more context to the log messages, such as the event type (i.e., insert or update), to make it clearer what actions are being performed. Also, ensure consistent capitalization of log messages.

8. Overall readability: Consider improving the formatting of the code as there are inconsistencies in indentations and spacing that can affect readability. Additionally, there are some unnecessary comments and commented-out code that can be removed.

9. Potential performance optimization: The code currently populates the `items` and `tabId` fields of the order object in both the insert and update cases. It may be possible to optimize performance by selectively populating only the necessary fields based on the operation type to reduce the amount of data retrieved from the database.